KsApi/extensions/ApolloInterceptors.swift

```diff
@@ -0,0 +1,43 @@
+import Apollo
+import Foundation
+
+// MARK: - NetworkInterceptorProvider
+
+class NetworkInterceptorProvider: LegacyInterceptorProvider {
```

Comment:
Ah so I'm unfamiliar with Interceptors. Are all requests the ApolloClient is performing going through the `async` method below?

Reply:
Interceptors are like hooks, just an API that Apollo provides to allow us to intercept requests and modify them if necessary. We need to do this in order to inject the PerimeterX headers because they could change between requests, or at least aren't known immediately at startup.

Reply:
> Yea why do we need this, can't we just pass in additional headers through `RequestChainNetworkTransport` in the singleton?

That would only be at instantiation and we don't know when these headers could change.

Reply:
Gotcha, that makes sense.